src/components/IGVFileSelector.js

```diff
@@ -83,41 +123,69 @@ export const getValidIgvFiles = (values) => {
       }
 
       // Filter to URLs that point to a file with one of the relevant extensions.
-      const basename = url.pathname.split('/').at(-1);
-      const [base, extension] = splitExtension(basename);
-      return !!base && allFiles.includes(extension);
+      const filename = url.pathname.split('/').at(-1);
+      const isValid = hasValidIgvExtension(filename);
+      return isValid;
     } catch (err) {
       return false;
     }
   });
 
+  const accessUrls = await resolveValidIgvDrsUris(values, signal);
+  const fileUrlStrings = basicFileUrls.concat(accessUrls);
+  const fileUrls = fileUrlStrings.map((fus) => new URL(fus));
+
   return fileUrls.flatMap((fileUrl) => {
-    if (fileUrl.endsWith('.bed')) {
-      return [{ filePath: fileUrl, indexFilePath: false }];
+    if (fileUrl.pathname.endsWith('.bed')) {
+      return [{ filePath: fileUrl.href, indexFilePath: false }];
     }
     const indexFileUrl = findIndexForFile(fileUrl, fileUrls);
-    return indexFileUrl !== undefined ? [{ filePath: fileUrl, indexFilePath: indexFileUrl }] : [];
+    if (indexFileUrl !== undefined) {
+      return [{ filePath: fileUrl.href, indexFilePath: indexFileUrl.href }];
+    }
+    return [];
   });
```

Comment:
I feel this would be more intuitive as an `if ... else if ... else` construct.

Reply:
I agree your suggestion would be a bit more intuitive.  Per chat, the current construct is due to the `no-else-return` ESLint rule.  We could disable it like so to get the `if ... else if ... else ...` construct, but opted to stay with project conventions:

```
return fileUrls.flatMap((fileUrl) => {
  if (fileUrl.pathname.endsWith('.bed')) {
    return [{ filePath: fileUrl.href, indexFilePath: false }];
    // eslint-disable-next-line no-else-return
  } else {
    const indexFileUrl = findIndexForFile(fileUrl, fileUrls);
    if (indexFileUrl !== undefined) {
      return [{ filePath: fileUrl.href, indexFilePath: indexFileUrl.href }];
      // eslint-disable-next-line no-else-return
    } else {
      return [];
    }
  }
});
```